File: projectNew/old_src/TTA/TTAActiveCardListener.java
package com.f14.TTA.listener;

import com.f14.TTA.component.card.TTACard;
import com.f14.TTA.component.param.RoundParam;
import com.f14.TTA.consts.TTAGameCmd;


/**
 * 激活卡牌能力的监听器
 *
 * @author F14eagle
 */
public abstract class TTAActiveCardListener extends TTAInterruptListener {
    protected RoundParam param;
    protected TTACard activeCard;

    public TTAActiveCardListener(RoundParam param, TTACard card) {
        super(param.player);
        this.param = param;
        this.activeCard = card;
    }

    @Override
    protected int getValidCode() {
        return TTAGameCmd.GAME_CODE_ACTIVABLE_CARD;
    }
}

File: projectNew/src/com/f14/TTA/listener/TTAActiveCardListener.kt
package com.f14.TTA.listener

import com.f14.TTA.component.card.TTACard
import com.f14.TTA.component.param.RoundParam
import com.f14.TTA.consts.TTAGameCmd

/**
 * Created by 吹风奈奈 on 2017/7/20.
 */

abstract class TTAActiveCardListener(protected var param: RoundParam, protected var activeCard: TTACard) : TTAInterruptListener(param.gameMode, param.player) {

    override val validCode: Int
        get() = TTAGameCmd.GAME_CODE_ACTIVABLE_CARD
}
